court/tests.py

```diff
@@ -0,0 +1,129 @@
+import pytest
+from court.models import Court
+from decimal import Decimal
+from django.db.utils import IntegrityError
+from django.core.exceptions import ValidationError
+
+
+@pytest.mark.django_db
+class TestCourtModel:
+
+    @pytest.fixture
+    def court(self):
+        court = Court.objects.create(x=Decimal('11'), y=Decimal('22'),
+                                     city='TEST_CITY', neighborhood='TEST_NEIGHBORHOOD', max_players=30)
+        return court
+
+    def test_get_instance(self, court):
+        court_in_db = Court.objects.get(pk=court.pk)
+        assert court_in_db == court
+
+    def test_create_court_with_existing_id(self, court):
+        with pytest.raises(IntegrityError):
+            Court.objects.create(courtID=1, x=Decimal('11'), y=Decimal('22'),
+                                 city='TEST_CITY', neighborhood='TEST_NEIGHBORHOOD', max_players=30)
+
+    def test_create_valid_court(self):
+        Court.objects.create(x=Decimal('11'), y=Decimal('22'),
+                             city='TEST_CITY', neighborhood='TEST_NEIGHBORHOOD', max_players=30).full_clean()
+
+    def test_create_three_courts_without_id(self):
+        court_id1 = Court.objects.create(x=Decimal('11'), y=Decimal('22'),
+                                         city='TEST_CITY', neighborhood='TEST_NEIGHBORHOOD', max_players=30)
+        court_id2 = Court.objects.create(x=Decimal('22'), y=Decimal('33'),
+                                         city='TEST_CITY', neighborhood='TEST_NEIGHBORHOOD', max_players=40)
+        court_id3 = Court.objects.create(x=Decimal('33'), y=Decimal('44'),
+                                         city='TEST_CITY', neighborhood='TEST_NEIGHBORHOOD', max_players=50)
+        assert court_id1.courtID == 1
```

Comment:
There's a built-in frugality in this sort of test as you assume unnecessary things about the environment. In this case, you assume the first one will get ID of 1, but suppose that something changes in the setup and now you start the test with a court ID already in the system, so your test gets broken.

I think it would have been better to create the first one, grab the ID and then assert that the next one is its plus 1.

Reply:
@yftacherzog I wasn't thinking of that option. You're right.
I'll open a new issue and fix it ASAP.